### DISCUSSION TITLE
Hot-reloading css for different extensions / using one-loader

### DISCUSSION THREAD

**Discussion by Zauberfisch** [react-static/react-static#1514] [Q&A]
**tl;dr: How can I specify what extensions the dev server watches and what triggers a style reload?** (I want changes in `foobar.one` to reload styles.

Long version: I'm planing to use [one-loader](https://github.com/digitalie/one-loader) with react-static, and so far I've got almost everything working with the following config:

```
// node.api.js
export default (options = {}) => {
	return {
		webpack: (webpackConfig, state) => {
			const {
				jsLoader,
				cssLoader,
			} = state.defaultLoaders;
			jsLoader.use[0].options.presets = [
				["react-static/babel-preset.js", {module: false}]
			];
			webpackConfig.resolve.extensions.push('.one');
			webpackConfig.module.rules[0].oneOf.unshift({
				test: /\.one$/,
				use: [
					{
						loader: 'one-loader',
						options: {
							map: {
								'text/css': cssLoader.loader,
								'text/scss': [...cssLoader.loader, 'sass-loader'],
								'javascript': jsLoader.use,
								'text/jsx': jsLoader.use,
							}
						}
					},
				]
			});
			return webpackConfig;
		},
	}
};
```
And an example .one file:
```
<script type="text/jsx">
	import React from 'react'

	export default () => (
		<div className="foobar">This is a <span>test</span></div>
	);

</script>
<style type="text/scss">
	.foobar {
		background: orange;
		color: red;

		span {
			background: red;
			color: orange;
		}
	}
</style>
```

Changes inside the `<script>` will be hot-reloaded. However, if I change something in `<style>` it will not reload.
But if I later change an actual css file like `App.css`, it will load both my changes to the App.css aswell as the .one file.

This leads me to believe that compiling actually works, but I need to tell the dev server to reload js and css if a .one file changes.
But I was unable to find that so far.
